File: Counting/3527.Find the Most Common Response/Solution.py
class Solution:
    def findCommonResponse(self, responses: List[List[str]]) -> str:
        mx = 1
        mx_w = []
        g = defaultdict(int)
        for r in responses:
            s = set(r)
            for w in s:
                g[w] += 1
                if mx < g[w]:
                    mx = g[w]
                    mx_w = []
                if mx == g[w]:
                    mx_w.append(w)
        mx_w.sort()
        return mx_w[0]

File: Counting/3527.Find the Most Common Response/Solution.java
class Solution {

    public String findCommonResponse(List<List<String>> responses) {
        int mx = 1;
        List<String> mxW = new ArrayList<>();
        Map<String, Integer> g = new HashMap<>();
        for (List<String> r : responses) {
            Set<String> s = new HashSet<>(r);
            for (String w : s) {
                g.put(w, g.getOrDefault(w, 0) + 1);
                if (mx < g.get(w)) {
                    mx = g.get(w);
                    mxW.clear();
                }
                if (mx == g.get(w)) {
                    mxW.add(w);
                }
            }
        }
        Collections.sort(mxW);
        return mxW.get(0);
    }
}
